frontend/src/libs/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,103 @@
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useCallback, useState } from '~/libs/hooks/hooks.js';
+
+import { Button } from '../button/button.jsx';
+import styles from './styles.module.scss';
+
+type Properties = {
+  pagesRange: number;
+  paginationLength?: number;
+  onClick: (n: number) => void;
+};
+
+const Pagination: React.FC<Properties> = ({
+  pagesRange,
+  paginationLength = pagesRange > 5 ? 5 : pagesRange,
+  onClick,
+}: Properties) => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const middleValue = Math.floor((paginationLength - 1) / 2);
+
+  const handlePageClick = useCallback(
+    (event_: React.MouseEvent) => {
+      const target = event_.target as HTMLButtonElement;
+      const index = target.textContent;
+
+      if (index) {
+        onClick(+index - 1);
+        setCurrentPage((currentPage) => (index ? +index : currentPage));
+      }
+    },
+    [onClick],
+  );
+  const handlePreviousClick = useCallback(() => {
+    onClick(+currentPage - 2);
+    setCurrentPage((currentPage) => currentPage - 1);
```

Comment:
Why -2 and -1?

Reply:
Inside onClick we should set the index page that’s why  -1  and we need to set the previous page index  -2. Should I refactor it?